db/migrate/20250902233414_add_scoring_method_to_problem.rb

```diff
@@ -0,0 +1,9 @@
+class AddScoringMethodToProblem < ActiveRecord::Migration
+  def change
+    add_column :problems, :scoring_method, :integer, default: 1 # 0: max-submission, 1: subtask-scoring
```

Comment:
This means that all new problems will default to subtask-scoring, just checking that's intentional

Reply:
Yes, I think it is reasonable as most new problems created would work fine with subtask scoring. I am open to changing it though.